doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py

```diff
@@ -3,16 +3,20 @@
 import numpy
 
 import Orange.data
-from Orange.widgets import widget, gui
+from Orange.widgets.widget import OWWidget, Input, Output
+from Orange.widgets import gui
 
-class OWDataSamplerA(widget.OWWidget):
+class OWDataSamplerA(OWWidget):
     name = "Data Sampler"
     description = "Randomly selects a subset of instances from the data set"
     icon = "icons/DataSamplerA.svg"
     priority = 10
 
-    inputs = [("Data", Orange.data.Table, "set_data")]
-    outputs = [("Sampled Data", Orange.data.Table)]
+    class Inputs:
+        data = Input("Data", Orange.data.Table)
+
+    class Outputs:
+        sample = Output("Sampled Data", Orange.data.Table)
```

Comment:
Understandably, it's a little late for severe API change proposals, but how do we feel about:
* `Inputs` and `Outputs` identifiers being fixed, and
* in OWWidget metaclass interpret `(attr, value)` properties of these classes such that:
  * if `value` is an `Input`, no change;
  * if `value` is a `dict`, fill `Input(**d)`;
  * else, set value to `Input(name=attr.title(), type=value)`.

The simpler definitions could then be:
```py
class Inputs:
    data = Table
class Outputs:
    sampled_data = Table
```

Deriving the idea from `OWWidget.Error` where messages must be instances of `Msg()` whereas simple strings would (with some help from the metaclass) suffice.